### ISSUE TITLE
.gitignore files do not show in file explorer

### ISSUE THREAD

**Issue by Crazypersonalph** [microsoft/vscode#152543]
Issue Type: <b>Bug</b>

Open any project that has been initialized with ```git init```, place a .gitignore file in it, and it will not show

I am running the latest Code-Insiders build (through the Arch Linux AUR package, ```visual-studio-code-insiders-bin```)

If you try to add a .gitignore file, it will say .gitignore file already exists

VS Code is detecting the .gitignore file, but it is not showing it

.gitignore is not in my Excludes list in the settings.json file

VS Code version: Code - Insiders 1.69.0-insider (fbf1cf3832d43088e27837dbb68d24ab65a098c1, 2022-06-17T05:17:59.312Z)
OS version: Linux x64 5.18.5-arch1-1
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-1195G7 @ 2.90GHz (8 x 2680)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 1, 1|
|Memory (System)|31.07GB (21.48GB free)|
|Process Argv|--no-sandbox --unity-launch --crash-reporter-id 534ad51f-3a50-4c19-83b9-fe3bf8f15214|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|plasma|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (29)</summary>

Extension|Author (truncated)|Version
---|---|---
npm-intellisense|chr|1.4.2
doxdocgen|csc|1.4.0
es7-react-js-snippets|dsz|4.4.3
gitlens|eam|12.1.1
vscode-npm-script|eg2|0.3.25
copilot|Git|1.30.6165
go|gol|0.33.1
cloudcode|goo|1.18.3
discord-vscode|icr|5.8.0
better-cpp-syntax|jef|1.15.18
cmake-language-support-vscode|jos|0.0.4
file-downloader|min|1.0.12
mindaro|min|1.0.120220125
vscode-docker|ms-|1.22.0
vscode-dotnet-runtime|ms-|1.5.0
vscode-kubernetes-tools|ms-|1.3.10
python|ms-|2022.8.0
vscode-pylance|ms-|2022.6.20
jupyter|ms-|2022.5.1001601848
remote-containers|ms-|0.238.2
remote-ssh|ms-|0.82.1
remote-wsl|ms-|0.66.3
cmake-tools|ms-|1.11.26
cpptools|ms-|1.10.7
cpptools-extension-pack|ms-|1.2.0
vscode-yaml|red|1.8.0
LiveServer|rit|5.7.5
svelte-vscode|sve|105.17.0
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256197
vslsvsres303:30308271
pythonvspyl392:30422396
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30404738
wslgetstarted:30449409
vscscmwlcmt:30465136
cppdebug:30492333
vscaat:30438846
pylanb8912:30496796
vsclangdf:30492506

```

</details>

**Comment by gjsjohnmurray** [comment]
Looks like a recent change means `.gitignore` now always gets hidden in Explorer unless you have set `"explorer.excludeGitIgnore": true`, in which case the .gitignore file shows but none of the `.gitignore`d files do.

@lszomoru this means that people who don't want to use the new `explorer.excludeGitIgnore` setting will no longer see their `.gitignore` file in Explorer. Is this intentional?

**Comment by Crazypersonalph** [comment]
> Looks like a recent change means `.gitignore` now always gets hidden in Explorer unless you have set `"explorer.excludeGitIgnore": true`, in which case the .gitignore file shows but none of the `.gitignore`d files do.
> 
> @lszomoru this means that people who don't want to use the new `explorer.excludeGitIgnore` setting will no longer see their `.gitignore` file in Explorer. Is this intentional?

It seems like setting that to true fixes the issue, but I do need to see the files in my .gitignore, such as .env files that contain environment variables.

Is there anyway I can see the .gitignore file, with it showing my excluded files?

**Comment by Crazypersonalph** [comment]
I do not think this is intentional, as I can see many issues this could arise because of this

**Comment by tjx666** [comment]
This is an issue...

**Comment by cwadrupldijjit** [comment]
I saw that even having a folder without Git initialized in it still results in the .gitignore being hidden.  Seems like it's a default behavior and the `explorer.excludeGitIgnore` setting changes this behavior.  I tried messing with the `Files: Exclude` setting and adding a negated glob (or whatever that's officially called) to get the gitignore back, but that didn't work.

**Comment by Crazypersonalph** [comment]
@lszomoru Any update on where this bug report is going?

**Comment by lszomoru** [comment]
Adding @lramos15 as he worked on this new functionality.

**Comment by lramos15** [comment]
> Issue has been fixed!

Glad to hear it! Sorry for the few bumps along the way :)
